python/tests/integration/arcticdb/version_store/test_snapshot.py

```diff
@@ -439,3 +423,46 @@ def test_remove_from_snapshot_multiple(basic_store_tombstone_and_pruning):
     assert len(versions) == 1
     assert lib.read("s3", as_of="saved").data == 3
     assert lib.read("s2", as_of="saved").data == 2
+
+
+def test_snapshot_not_accept_tombstoned_key(basic_store_delayed_deletes_v1, sym):
+    lib = basic_store_delayed_deletes_v1
+    ver = lib.write(sym, 1).version
+    lib.write(sym, 2).version
+    with pytest.raises(NoSuchVersionException):
+        lib.snapshot("s", versions={sym:ver})
+
+
+def test_snapshot_partially_valid_version_map(basic_store_delayed_deletes_v1):
+    lib = basic_store_delayed_deletes_v1
+    symA = "A"
+    symB = "B"
+
+    symA_ver1 = lib.write(symA, 1).version
+    lib.write(symA, 2).version
```

Comment:
This `write` is just being used to delete version 0 via prune previous, would be clearer to just call `delete`.